packages/cactus-plugin-keychain-vault/src/test/typescript/unit/transit-vault.test.ts

```diff
@@ -0,0 +1,188 @@
+    transitSecretMountPath: mountPath,
+    instanceId: uuidv4(),
+  };
+
+  t.test("newKey", (t: Test) => {
+    const client = new PluginKeychainVault(options);
+    {
+      // new key of type EC256
+      t.ok(client.transitNewKey(testECP256, VaultTransitKeyType.EC256));
+    }
+    {
+      // new key of type EC384
+      t.ok(client.transitNewKey(testECP384, VaultTransitKeyType.EC384));
+    }
+    {
+      // new key of type EC521
+      t.ok(client.transitNewKey(testECP521, VaultTransitKeyType.EC521));
+    }
+    t.end();
+  });
+  t.test("sign", async (t: Test) => {
+    const client = new PluginKeychainVault(options);
+
+    const digest = Buffer.from("helloCactus");
+    const hashedDigest = createHash("sha256").update(digest).digest();
+
+    {
+      // sign with EC 256
+      const signature = await client.transitSign(testECP256, digest, false);
+      t.ok(signature);
+    }
+    {
+      // sign with EC 384 and digest is already hashed
+      const signature = await client.transitSign(
+        testECP384,
+        hashedDigest,
+        true,
+      );
+      t.ok(signature);
+    }
+
+    {
+      // sign with EC 521 and digest is already hashed
+      const signature = await client.transitSign(
+        testECP521,
+        hashedDigest,
+        true,
+      );
+      t.ok(signature);
+    }
+
+    {
+      // sign with non existing key
+      t.test("sign with non existing key", async (t: Test) => {
+        try {
+          await client.transitSign("not-found", hashedDigest, true);
+        } catch (error) {
+          t.equal("encryption key not found", (error as Error).message);
+        }
+        t.end();
+      });
+    }
+    t.end();
+  });
+
+  t.test("getPubKey", async (t: Test) => {
+    const client = new PluginKeychainVault(options);
+
+    {
+      // get pub key of type EC256
+      const pub = await client.transitGetPub(testECP256);
+      t.equal(pub.type, "public");
+      t.equal(pub.asymmetricKeyDetails?.namedCurve, "prime256v1");
+      t.equal(pub.asymmetricKeyType, "ec");
+    }
+
+    {
+      // get pub key of type EC384
+      const pub = await client.transitGetPub(testECP384);
+      t.equal(pub.type, "public");
+      t.equal(pub.asymmetricKeyDetails?.namedCurve, "secp384r1");
+      t.equal(pub.asymmetricKeyType, "ec");
+    }
+
+    {
+      // get pub key of type EC521
+      const pub = await client.transitGetPub(testECP521);
+      t.equal(pub.type, "public");
+      t.equal(pub.asymmetricKeyDetails?.namedCurve, "secp521r1");
+      t.equal(pub.asymmetricKeyType, "ec");
+    }
+
+    {
+      // get non existing key
+      try {
+        await client.transitGetPub("not-found");
+      } catch (error) {
+        t.equal("key not found", (error as Error).message);
+      }
+    }
+    t.end();
+  });
+
+  t.test("rotateKey", async (t: Test) => {
+    const client = new PluginKeychainVault(options);
+
+    {
+      // rotate EC256 key
+      await client.transitRotateKey(testECP256);
```

Comment:
@Zzocker Is there any way to verify that the rotation has indeed happened? I'm thinking maybe we could perform some signing operation after the rotation and verify that the signature was done with the new key.
Please do let me know if this is a huge rabbit hole, then we can reevaluate it. If it's not too much trouble then it would be really nice to have some assertions like above.

Reply:
I have already added this case inside integration test. [here](https://github.com/Zzocker/cactus/blob/bb31bb73bc58126ad30657eb0281c95499efd822/packages/cactus-plugin-keychain-vault/src/test/typescript/integration/plugin-keychain-vault.test.ts#L429)

```ts
      // check if signature is signed with new key
      const verify = createVerify("sha256");
      verify.update(digest);
      t.false(verify.verify(pub, sigRotated), "signature should not be valid");
```

This test is to check whether signature is signed with rotated key or previous key, This test works like once key has been rotated the generated signature should be valid against previous key  .